src/main/java/net/mcreator/element/types/Block.java

```diff
@@ -387,4 +393,14 @@ public TextureHolder getParticleTexture() {
 		return particleTexture == null || particleTexture.isEmpty() ? texture : particleTexture;
 	}
 
+	public static class PropertyEntry {

```

Comment:
It relates to your question (where you ask it if is okish, that we need two identical methods to set value, which comes back to this), but this still raises questions to me. It just is not Javanic with this approach.

Looking at LivingEntity, this is handled as `public List<PropertyDataWithValue<?>> entityDataEntries;`.

I really don't think you should abuse data model structure for sake of utility methods

Reply:
Where should the utility methods be moved in this case?

Reply:
IIRC it is two methods. What will be signature and purpose of this methods? Maybe this will help us find the most optimal solution

Reply:
> one added now

This one would not be needed if PropertyEntry is ommited

> one checks if property is of text type (to put quotation marks around their values)

As you said, this could be a part of FTL

> another converts snake-case registry name to camel-case class name for string properties.

For this, we can expose StringUtils to FTL

------

all of these changes are less problematic than hacking data model for util methods

Reply:
> For this, we can expose StringUtils to FTL

This is the idea I had in mind recently. Fine to do this in this PR, in another small PR, or would you do this?